directord/utils.py

```diff
@@ -613,37 +620,82 @@ def clear(self):
         for item in self.keys():
             self.__delitem__(key=item)
 
+    def size(self):
+        """Return the current cache size."""
+
+        count = 0
+        for _ in self.keys():
+            count += 1
+
+        return count
+
 
-class DurableQueue(queue.Queue):
+class DurableQueue:
     """Durable queue class, used to ensure queued items are disk backed."""
 
-    def __init__(self, maxsize, mutex, lock, condition, path):
-        self.maxsize = maxsize
-        self.mutex = mutex
-        self.not_empty = condition(self.mutex)
-        self.not_full = condition(self.mutex)
-        self.all_tasks_done = condition(self.mutex)
+    def __init__(self, mutex, fslock, path):
         self.path = path
-        self.queue = Cache(url=self.path, lock=lock)
-        self.unfinished_tasks = self._qsize()
+        self.mutex = mutex
+        self.queue = Cache(url=self.path, lock=fslock)
 
-    def _init(self, *args, **kwargs):
-        pass
+    def _get(self):
+        """Retrieve the first item from the queue."""
 
-    def _qsize(self):
-        """Return the queue size."""
-        return len(list(self.queue.keys()))
+        for item in self.queue.keys():
+            return self.queue.pop(item)
+        else:
+            raise queue.Empty
 
     def _put(self, item):
         """Put a new item within the queue."""
+
         self.queue[get_uuid()] = item
 
-    def _get(self):
-        """Retrieve the first item from the queue."""
-        for item in self.queue.keys():
-            return self.queue.pop(item)
+    def _qsize(self):
+        """Return the queue size."""
+
+        try:
+            return self.queue.size()
+        except Exception:
+            return 0
+
+    def join(self):
+        """Join the current Queue and cleanup artifacts."""
+
+        if not self.mutex.aquire(blocking=True, timeout=None):
```

Comment:
just to double-check, are you sure it should ignore the blocking/timeout options for this (only?) special place?

Reply:
yeah this was mirrored from the simple queue implementation. All that said, I think I have to go back to the drawing board for the rest of the durable queue implementation.